packages/ai/src/Button.ts

```diff
@@ -58,13 +87,185 @@ class Button extends UI5Element {
 	@property({ type: Boolean })
 	disabled!: boolean;
 
+	/**
+	 * Defines the current state of the component.
+	 *
+	 * **Note:** if nothing is defined, the component will be set initially to the first defined state (if any).
+	 *
+	 * @default ""
+	 * @public
+	 */
 	@property()
-	icon!: string;
+	state?: string;
 
-	@property()
-	endIcon!: string;
+	/**
+	 * Keeps the current state of the component.
+	 * @private
+	 */
+	@property({ type: Object })
+	_currentStateObject?: ButtonState;
+
+	/**
+	 * Initiates button elements fade-out phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeOut!: boolean;
+
+	/**
+	 * Initiates button fade middle phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeMid!: boolean;
+
+	/**
+	 * Initiates button elements fade-in phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeIn!: boolean;
+
+	/**
+	 * Defines the available states of the component.
+	 * **Note:** Although this slot accepts HTML Elements, it is strongly recommended that you only use `ui5-ai-button-state` components in order to preserve the intended design.
+	 * @public
+	 */
+	@slot({ type: ButtonState, "default": true })
+	states!: Array<ButtonState>;
+
+	onBeforeRendering(): void {
+		if (this.fadeOut || this.fadeIn) {
+			return;
+		}
+
+		if (this.state && this._currentStateObject && !Object.keys(this._currentStateObject).length) {
+			this._currentStateObject = this._findStateByName(this.state);
+		}
+
+		const currentStateName = this._currentStateObject?.name || "";
+
+		if (!this.state) {
+			this.state = this.states.length ? this.states[0].name : "";
+			this._currentStateObject = this._findStateByName(this.state);
+		}
+
+		if (!this._currentStateObject) {
+			this._throwMissingStateError();
+		}
+
+		if (currentStateName !== "" && currentStateName !== this.state) {
+			this._fadeOut();
+		}
+	}
+
+	onAfterRendering(): void {
+		if (!this._findStateByName(this.state!) && this._currentStateObject) {
+			this.state = this._currentStateObject.name;
+		}
+	}
+
+	/**
+	 * Starts the fade-out animation.
+	 * @private
+	 */
+	async _fadeOut(): Promise<void> {
+		const button = this.shadowRoot?.querySelector("[ui5-button]") as MainButton;
+		const newStateObject = this._findStateByName(this.state!);
+		if (button && newStateObject) {
+			const buttonWidth = (button as HTMLElement).offsetWidth;
```

Comment:
does .offsetWidth require the cast? We have MainButton which inherits UI5Element, which inherits HTMLElement, so the cast shouldn't be needed.

Reply:
Yes, you're right, fixed.